lib/src/mobile_scanner_controller.dart

```diff
@@ -171,6 +177,21 @@ class MobileScannerController extends ValueNotifier<MobileScannerState> {
 
           value = value.copyWith(zoomScale: zoomScale);
         });
+
+    if (defaultTargetPlatform != TargetPlatform.macOS) {
+      if (MobileScannerPlatform.instance
+          case final MethodChannelMobileScanner implementation) {
```

Comment:
Move the outer if to the when clause

```suggestion
          case final MethodChannelMobileScanner implementation when defaultTargetPlatform != TargetPlatform.macOS) {
```